src/goto-synthesizer/CMakeLists.txt

```diff
@@ -0,0 +1,25 @@
+# Library
+file(GLOB_RECURSE sources "*.cpp" "*.h")
+list(REMOVE_ITEM sources
+    ${CMAKE_CURRENT_SOURCE_DIR}/goto_synthesizer_main.cpp
+)
+add_library(goto-synthesizer-lib ${sources})
+
+generic_includes(goto-synthesizer-lib)
+
+target_link_libraries(goto-synthesizer-lib
+    ansi-c
+    cpp
+    big-int
+    goto-checker
+    goto-instrument-lib
+    goto-programs
+    langapi
+    util
+    json
+)
+
+# Executable
+add_executable(goto-synthesizer goto_synthesizer_main.cpp)
+target_link_libraries(goto-synthesizer goto-synthesizer-lib)
+install(TARGETS goto-synthesizer DESTINATION ${CMAKE_INSTALL_BINDIR})
```

Comment:
Can you please also add a man page (in doc/man/) even when it will basically be empty at this point? And then please add an `install` rule for it in here plus amend `scripts/check_help.sh` to include the new binary.